privacy/scrubber.go

```diff
@@ -147,7 +147,10 @@ func (scrubber) ScrubRequest(bidRequest *openrtb2.BidRequest, enforcement Enforc
 	if enforcement.TID {
 		//remove source.tid and imp.ext.tid
 		if bidRequest.Source != nil {
-			bidRequest.Source.TID = ""
+			var sourceCopy *openrtb2.Source
+			sourceCopy = ptrutil.Clone(bidRequest.Source)
```

Comment:
i think it's possible to remove the var declaration on line 150, and just have this line instead `sourceCopy := ptrutil.Clone(bidRequest.Source)`

Reply:
This may be the case, I think `Clone()`, while being a generic, will get the pointer type from the type if the argument used. That it, whatever pointer argument you feed it, it will return the same pointer type.
```
func Clone[T any](v *T) *T 
```
The provided type, a pointer to  a `T` type, indicates that the return value should be another pointer to type `T`.

Reply:
Yup. I'd expect `sourceCopy := ptrutil.Clone(bidRequest.Source)` to work here.

Reply:
I tried it and it still needs some type conversion. Based on other usages of "Clone" in code, type is declared beforehand. I may miss something, please advice.

Reply:
Darn. Go's generic features are still pretty new. I was hoping this syntax would infer the generic type from the `bidRequest.Source`, but it this didn't work then it would seem the compiler is inferring from the variables type to align the return type of `Clone`.